soc/nordic/common/uicr/gen_uicr/CMakeLists.txt

```diff
@@ -62,7 +64,12 @@ if(CONFIG_GEN_UICR_GENERATE_PERIPHCONF)
     endif()
 
     if(EXISTS ${_dir}/zephyr/zephyr.dts)
-      list(APPEND periphconf_elfs ${_dir}/zephyr/zephyr.elf)
+      # Check if this is secondary firmware by looking for the marker file
+      if(EXISTS ${_dir}/is_secondary_firmware.txt)
+        list(APPEND secondary_periphconf_elfs ${_dir}/zephyr/zephyr.elf)
+      else()
+        list(APPEND periphconf_elfs ${_dir}/zephyr/zephyr.elf)
+      endif()
```

Comment:
Maybe there is a way to distinguish between primary/secondary at a lower level, even including that information in the ELF itself? For example, if we defined two different periphconf sections in the elf instead of just one, and chose which to populate based on some config in the image (changing where we put things with the macros in uicr.h). Then each input file would be self describing, and this part wouldn't need a marker file.

Reply:
I'm sure that is possible.

But it seems a lot more complicated than just having a text file that signifies if an image is a secondary image or not.

It is easier both for humans and the build system to create and read files than it is for humans and build systems to modify and read elf files.